src/pages/home/home.ts

```diff
@@ -436,6 +438,11 @@ export class HomePage {
             handler: () => {
               this.groupSocketService.leaveGroup();
               this.isInGroup = false;
+              this.isSpaceCreated = false;
+              this.selectedLocation = null;
+              for (var i = 1; i < this.gmarkers.length; i++) {
```

Comment:
What is exactly happening here? We are removing all the contents of `gmarkers` except for the first one?

Reply:
Yes, the first one to be added to the array is the current location and that's the one we want to keep.

Reply:
There's definitely an easier way to do this, check this out: https://stackoverflow.com/a/39510882/4851234

Reply:
I'm actually calling the .remove() on each marker in the array, not actually removing them from the array, sorry I worded it a bit ambiguously.